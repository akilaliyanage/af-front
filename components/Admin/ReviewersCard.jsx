import React,{useState,useEffect} from "react";
import '../../assets/css/admin/admin.css'
import axios from "axios";
import {Link} from "react-router-dom";
import config from "../../config.json";
import { BsPencilSquare } from 'react-icons/bs';

function ReviewersCard(props){


    const [reviewData,setReviewers] = useState([]);


    useEffect(() => {
        const url = config.host+"/reviewer";
        axios.get(url).then((res) => {
            setReviewers(res.data);
        }).catch((err) => {
            console.log(err);
        })

    },[]);


    return(

        <div className="uditha-card uditha-reviewer-card-align">

        <div className="uditha-reviewers">

            <h4 style={{color:"white"}}>Current Reviewers</h4>
            <div className="uditha-reviewer-button-align">
            <button onClick={e =>  window.location.href='/add-reviewer'} className="button-add-reviewer">Add new</button>
            </div>
            <br/>
            {
                reviewData.map((reviewer) =>

                    <div>

                        <table style={{padding:"5px",borderSpacing:"1px",tableLayout:"fixed",width:"400px"}}>
                            <tr>

                                <th><img src={config.host+"/"+reviewer.profileImage} width="70"  style={{"border-radius": "50%"}}/> </th>
                                <td style={{color:"white"}}>{reviewer.name}</td>

                                <Link to ={`update-reviewer/${reviewer._id}`}>
                                <td><BsPencilSquare/></td>
                                </Link>
                            </tr>

                        </table>





                    </div>


            )
            }



        </div>

        </div>

)
}

export default ReviewersCard;
