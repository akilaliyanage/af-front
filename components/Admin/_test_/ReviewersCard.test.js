import React  from 'react';
import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import expectExport from 'expect';
import ReviewersCard from '../ReviewersCard';

afterEach(()=>cleanup())


it('Compare the snapshot of Reviewers card', () =>{
    const tree = renderer.create(
        <ReviewersCard/>
    ).toJSON();
    expectExport(tree).toMatchSnapshot();
})


test('Make sure component render without any issue', () => {
    render(<ReviewersCard />);
    screen.debug();
    expect(screen.getByText('Add new')).toBeInTheDocument();
  });