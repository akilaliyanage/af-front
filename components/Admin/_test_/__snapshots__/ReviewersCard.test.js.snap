// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare the snapshot of Reviewers card 1`] = `
<div
  className="uditha-card uditha-reviewer-card-align"
>
  <div
    className="uditha-reviewers"
  >
    <h4
      style={
        Object {
          "color": "white",
        }
      }
    >
      Current Reviewers
    </h4>
    <div
      className="uditha-reviewer-button-align"
    >
      <button
        className="button-add-reviewer"
        onClick={[Function]}
      >
        Add new
      </button>
    </div>
    <br />
  </div>
</div>
`;

exports[`create a snapshot 1`] = `
<div
  className="uditha-card uditha-reviewer-card-align"
>
  <div
    className="uditha-reviewers"
  >
    <h4
      style={
        Object {
          "color": "white",
        }
      }
    >
      Current Reviewers
    </h4>
    <div
      className="uditha-reviewer-button-align"
    >
      <button
        className="button-add-reviewer"
        onClick={[Function]}
      >
        Add new
      </button>
    </div>
    <br />
  </div>
</div>
`;

exports[`create a snapshot of Reviewers card 1`] = `
<div
  className="uditha-card uditha-reviewer-card-align"
>
  <div
    className="uditha-reviewers"
  >
    <h4
      style={
        Object {
          "color": "white",
        }
      }
    >
      Current Reviewers
    </h4>
    <div
      className="uditha-reviewer-button-align"
    >
      <button
        className="button-add-reviewer"
        onClick={[Function]}
      >
        Add new
      </button>
    </div>
    <br />
  </div>
</div>
`;
